--- a/dwm.c	2012-02-08 21:39:01.383020439 +0200
+++ b/dwm.c	2012-02-08 21:39:01.385020439 +0200
@@ -868,6 +868,8 @@ enternotify(XEvent *e) {
 	Monitor *m;
 	XCrossingEvent *ev = &e->xcrossing;
 
+	if(clicktofocus)
+		return;
 	if((ev->mode != NotifyNormal || ev->detail == NotifyInferior) && ev->window != root)
 		return;
 	c = wintoclient(ev->window);
@@ -1253,6 +1255,8 @@ motionnotify(XEvent *e) {
 	Monitor *m;
 	XMotionEvent *ev = &e->xmotion;
 
+	if(clicktofocus)
+		return;
 	if(ev->window != root)
 		return;
 	if((m = recttomon(ev->x_root, ev->y_root, 1, 1)) != mon && mon) {
